### PR TITLE
Added a placeholder tab check before creating a new scene.

### LINKED ISSUE #64836: Placeholder scene tab is not removed when the user creates a new scene
### Godot version

4.0 (#c8c0f21)

### System information

Windows 10, Ryzen 5900 X, Radeon 5700 XT

### Issue description

What the title says. On the surface, this is a minor issue, but in practice it results in the Scene dock not displaying any nodes for the current scene.

### Steps to reproduce

1. Open the editor and close all the scenes, if you have any open.
2. A placeholder scene tab appears.
3. Try opening an existing scene and witness it replacing the placeholder tab.
4. Close it and instead create a new scene.
5. Witness it adding a new tab, but the placeholder tab still remains.
6. The Scene dock correctly displays the root node of the freshly created scene.
7. Switch to the placeholder tab and either close it or just switch back.
8. Witness that the Scene dock is empty.

Switching between scene tabs otherwise doesn't result in this bug. Even if you open a new empty tab and switch back and forth, it doesn't break.

Here's a clip:

https://user-images.githubusercontent.com/11782833/186456346-a17b02aa-96a0-428a-a391-91a0ef33ee12.mp4

### Minimal reproduction project

N/a

### PR DESCRIPTION
**Pull request by faisal-alam09**
Fixes #64836
Added a Placeholder scene check before creating new scene .

### CODE REVIEW

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -3702,6 +3702,18 @@ void EditorNode::fix_dependencies(const String &p_for_file) {
 
 int EditorNode::new_scene() {
 	int idx = editor_data.add_edited_scene(-1);
+	// Remove [empty] scene if any other scene is loaded.
+	if (editor_data.get_edited_scene_count() > 1) {
+		for (int i = 0; i < editor_data.get_edited_scene_count() - 1; i++) {
+			bool unsaved = get_undo_redo()->is_history_unsaved(editor_data.get_scene_history_id(i));
+			if (!unsaved && ((*editor_data.get_scene_path(i).get_data()) == 0)) {
```
Comment (KoBeWi) [inline editor/editor_node.cpp:3709]:
```suggestion
			if (!unsaved && editor_data.get_scene_path(i).get_data().is_empty()) {
```
This is how you should check for empty string.

File: editor/editor_node.cpp [hunk 2]
```diff
@@ -3702,6 +3702,18 @@ void EditorNode::fix_dependencies(const String &p_for_file) {
 
 int EditorNode::new_scene() {
 	int idx = editor_data.add_edited_scene(-1);
+	// Remove [empty] scene if any other scene is loaded.
```
Comment (KoBeWi) [inline editor/editor_node.cpp:3705]: The [empty] is also removed if there is another [empty] (so no scene loaded), so the comment is not exactly correct. Maybe something like "Remove placeholder empty scene." would be better.

**Comment by KoBeWi** [comment]
You will also need to squash your commits. See here: https://docs.godotengine.org/en/stable/community/contributing/pr_workflow.html#the-interactive-rebase

File: editor/editor_node.cpp (hunk 1 above)
Comment (faisal-alam09) [reply editor/editor_node.cpp:3709]: But ,
editor_data.get_scene_path(i).get_data() is of type = const char32_t * ...

Comment (KoBeWi) [reply editor/editor_node.cpp:3709]: Sorry, typo.

File: editor/editor_node.cpp (hunk 2 above)
Comment (faisal-alam09) [reply editor/editor_node.cpp]: one single commit , Done.
rewriting comment , Done.

File: editor/editor_node.cpp (hunk 1 above)
Comment (KoBeWi) [reply editor/editor_node.cpp:3709]: This is still unresolved, I edited my suggestion.

Comment (faisal-alam09) [reply editor/editor_node.cpp:3709]: sorry ,
i haven't noticed that .

File: editor/editor_node.cpp [hunk 3]
```diff
@@ -3702,6 +3702,18 @@ void EditorNode::fix_dependencies(const String &p_for_file) {
...
+	if (editor_data.get_edited_scene_count() > 1) {
+		for (int i = 0; i < editor_data.get_edited_scene_count() - 1; i++) {
+			bool unsaved = get_undo_redo()->is_history_unsaved(editor_data.get_scene_history_id(i));
+			if (!unsaved && editor_data.get_scene_path(i).is_empty()) {
+				editor_data.remove_scene(i);
+				idx = idx - 1;
+			}
+		}
+	}
+	idx = idx < 0 ? 0 : idx;
```
Comment (KoBeWi) [inline editor/editor_node.cpp:3715]: Could be replaced with
```suggestion
	idx = MAX(idx, 0);
```

File: editor/editor_node.cpp [hunk 4]
```diff
@@ -3702,6 +3702,18 @@ void EditorNode::fix_dependencies(const String &p_for_file) {
 
 int EditorNode::new_scene() {
 	int idx = editor_data.add_edited_scene(-1);
+	// Remove placeholder empty scene.
+	if (editor_data.get_edited_scene_count() > 1) {
+		for (int i = 0; i < editor_data.get_edited_scene_count() - 1; i++) {
+			bool unsaved = get_undo_redo()->is_history_unsaved(editor_data.get_scene_history_id(i));
+			if (!unsaved && editor_data.get_scene_path(i).is_empty()) {
+				editor_data.remove_scene(i);
+				idx = idx - 1;
```
Comment (KoBeWi) [inline editor/editor_node.cpp:3711]: Could be
```suggestion
				idx -= 1;
```
or
```suggestion
				idx--;
```

File: editor/editor_node.cpp (hunk 1 above)
Comment (faisal-alam09) [reply editor/editor_node.cpp]: Changes Done .

File: editor/editor_node.cpp (hunk 3 above)
Comment (faisal-alam09) [reply editor/editor_node.cpp:3715]: okk

File: editor/editor_node.cpp (hunk 4 above)
Comment (faisal-alam09) [reply editor/editor_node.cpp:3711]: ok

File: editor/editor_node.cpp (hunk 3 above)
Comment (faisal-alam09) [reply editor/editor_node.cpp]: Done

File: editor/editor_node.cpp (hunk 4 above)
Comment (faisal-alam09) [reply editor/editor_node.cpp]: Done

**Review by KoBeWi — APPROVED** [review]
Seems ok.

As a last thing, you could edit your PR comment (on GitHub) to properly use a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**